rust/lance/src/index.rs

```diff
@@ -209,6 +209,28 @@ impl DatasetIndexExt for Dataset {
                 build_vector_index(self, column, &index_name, &index_id.to_string(), vec_params)
                     .await?;
             }
+            // Can't use if let Some(...) here because it's not stable yet.
+            // TODO: fix after https://github.com/rust-lang/rust/issues/51114
+            (IndexType::Vector, name)
+                if self.session.vector_index_extensions.contains_key(name) =>
```

Comment:
session.index_extensions? 
likely we dont want `self.session.{vector|scalar|fts}_extension` i guess?

Reply:
we will have them. We only have vector extensions for now. I can add these maps but they will just be empty